tests/test_edgeql_ddl.py

```diff
@@ -13085,6 +13085,25 @@ async def test_edgeql_ddl_create_migration_04(self):
         #     ['test'],
         # )
 
+    async def test_edgeql_ddl_create_migration_05(self):
+        await self.con.execute('''
+            create type X { create property x -> str; };
+        ''')
+        async with self.assertRaisesRegexTx(
+                edgedb.InvalidReferenceError,
+                "property 'x' does not"):
+            await self.con.execute('''
+                CREATE MIGRATION
+                {
+                    alter type default::X {
+                        alter property x rename to y;
+                    };
+                    alter type default::X {
+                        alter property x create constraint exclusive;
+                    };
+                };
+            ''')
```

Comment:
Comments mention a concrete constraint referenced by a short name, based on a renamed abstract constraint.

Can you add a test for that?

Reply:
That's tested by `test_schema_migrations_equivalence_annotation_08`, which was added in https://github.com/edgedb/edgedb/pull/4007, which introduced this bug as a regression. My first take at fixing it broke that one, which is how I realized it was a regression